basil/utils/utils.py

```diff
@@ -45,6 +46,61 @@ def bitarray_to_byte_array(bitarr):
     return array('B', bs.astype(np.uint8))
 
 
+def get_si_tcp(rel_path=''):
+    ''' Download SiTCP/SiTCP10G sources from the official github repo and apply patches
+    '''
+
+    sitcp_repo = r'https://github.com/BeeBeansTechnologies/SiTCP_Netlist_for_Kintex7'
+    sitcp_10G_repo = r'https://github.com/BeeBeansTechnologies/SiTCPXG_Netlist_for_Kintex7'
+
+    def line_prepender(filename, line):
+        with open(filename, 'rb+') as f:
+            content = f.read()
+            f.seek(0, 0)
+            add = bytearray()
+            add.extend(map(ord, line))
+            add.extend(map(ord, '\n'))
+            f.write(add + content)
+
+    sitcp_folder = os.path.join(os.path.os.getcwd(), rel_path, 'SiTCP/')
+
+    # Only download if the SiTCP git repository is not present
+    if not os.path.isdir(os.path.join(sitcp_folder, '.git')):
+        print('Downloading SiTCP')
+
+        # Has to be moved to be allowed to use existing folder for git checkout
+        git.Repo.clone_from(url=sitcp_repo,
+                            to_path=sitcp_folder, branch='master')
+        # Patch sources, see README of bdaq53
+        line_prepender(filename=sitcp_folder + 'TIMER.v', line=r'`default_nettype wire')
+        line_prepender(filename=sitcp_folder + 'WRAP_SiTCP_GMII_XC7K_32K.V', line=r'`default_nettype wire')
+        for line in fileinput.input([sitcp_folder + 'WRAP_SiTCP_GMII_XC7K_32K.V'], inplace=True):
+            print(line.replace("assign\tMY_IP_ADDR[31:0]\t= (~FORCE_DEFAULTn | (EXT_IP_ADDR[31:0]==32'd0) \t? DEFAULT_IP_ADDR[31:0]\t\t: EXT_IP_ADDR[31:0]\t\t);",
+                               'assign\tMY_IP_ADDR[31:0]\t= EXT_IP_ADDR[31:0];'), end='')
+    else:  # update if existing
+        print('SiTCP already present. Checking for updates')
+        g = git.cmd.Git(sitcp_folder)
+        g.pull()
+
+    sitcp_10G_folder = os.path.join(os.path.os.getcwd(), rel_path, 'SiTCP10G/')
+    # Only download if the SiTCP10G git repository is not present
+    if not os.path.isdir(os.path.join(sitcp_10G_folder, '.git')):
+        print('Downloading SiTCP10G')
+
+        # Has to be moved to be allowed to use existing folder for git checkout
+        git.Repo.clone_from(url=sitcp_10G_repo,
+                            to_path=sitcp_10G_folder, branch='master')
+        # Patch sources, see README of bdaq53
+        for line in fileinput.input([sitcp_10G_folder + 'WRAP_SiTCPXG_XC7K_128K.v'], inplace=True):
+            print(line.replace("\t\t.MY_IP_ADDR	\t\t\t\t\t(MY_IP_ADDR[31:0]	\t\t\t),\t// in\t: My IP address[31:0]",
+                               "\t\t.MY_IP_ADDR	\t\t\t\t\t({8'd192, 8'd168, 8'd100, 8'd12}),\t// in\t: My IP address[31:0]"), end='')
```

Comment:
The 10G flavor doesn't provide an input port for the IP address by default. We could add this feature with a few additional line replacements if that's something we need now or soon.
In general, it's very unlikely for the 10G implementation to be integrated in a network infrastructure (switches, routers etc.) because of how rare 10G networks still are. I would rather propose to postpone this feature until it's really needed.

Reply:
I was just curious, because that seems (and apparently is) inconsistent in SiTCP, but fair point to keep it as it is for now. The change is only minimal if really needed